### DISCUSSION TITLE
Some minor issues with the VDH version 9 beta

### DISCUSSION THREAD

**Discussion by GrampaWildWilly** [aclap-dev/video-downloadhelper#498] [Help and Q&A]
### Your question

These are things I noticed in the course of posting my response to #497.

In the report of an inflight download, it would be nice to show the accumulating size of the downloaded file.  This was something VDH 8.x used to give us.

I am going to complain again about the italic bold all caps entries on the VDH menu.  These should be not bold, not italic, simple mixed case text strings.  In addition, past releases displayed the entries on multiple lines if necessary.  You should bring that back.  If it takes 2 or 3 or 4 lines to show the full title of the video, so be it.  I asked for that.  Give it to me.  Which is to say, that used to be a VDH setting.  Please bring it back.  The string reported in the VDH menu should be identical to the file name proposed in the file selection dialog when you eventually launch the download.  That would be mixed case, not bold, not italic, & depending on the restoration of the missing setting, the entire string.

In conjunction with multi-line menu entries, you should give us a setting that allows us to set the width of the VDH menu.  The setting for making the "popup" larger isn't enough.  There used to be a setting in VDH 6.x.  You used to be able to control the width of the VDH menu by supplying a number of pixels.  With the possibility of multi-line entries, this old setting becomes very useful.  Please bring it back.

I do wish you would use the term "menu" instead of "popup."

The most recently used directory is actually not working.  When you launch a Download As, the directory VDH initially proposes is the one in the setting named Download directory.  But VDH has always updated that value with each download.  I don't necessarily want all my downloads to go in that directory.  When I download into a different directory, I want my next download to go into that new directory I just chose.  The value of that setting should change with each download, same as VDH always used to do.

### Addon & CoApp version and details

Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.13a1 beta, CoApp 2.0.19

**Comment by paulrouget** [comment]
Gonna summarize that:

**Missing Accumulating Download Size**: The current version does not show the accumulating size of the downloaded file in the inflight download report, which was a feature in VDH 8.x.

Added to todolist.

**Menu Text Formatting**: The VDH menu entries are currently displayed in italic bold all caps. These should be changed to simple mixed-case text strings without bold or italic formatting.

Let me try to experiment with that see how I like it.

**Multi-line Menu Entries**: Previous versions displayed long entries on multiple lines if necessary. This feature should be restored to accommodate long titles. 

That would potentially mess up the layout by making the hits bigger. We could maybe allow 2 lines.

**Menu Entry and File Name Consistency**: The string in the VDH menu should match the file name proposed in the file selection dialog (mixed case, not bold, not italic).

Titles are always more accurate than the filename. Some people have smartnaming rules that obfuscate the video title on the harddrive. We can't rely on the filename.

**Menu Width Setting**: There should be a setting to control the width of the VDH menu by specifying a number of pixels, similar to what was available in VDH 6.x.

That's doable.

**Terminology Change**: The term "popup" should be replaced with "menu."

Popup is the standard name. It's how it's called by all browsers vendor, and how it's used in documentation.

**Most Recently Used Directory Issue**: The "Download As" function does not remember the most recently used directory. The directory should update with each download instead of always defaulting to the initial setting named Download directory.

Right. So that's something I'm not sure what to do.

Now, it just uses whatever is set in the download directory setting. I didn't want to remember the directory because some people use Download As to specifically download in a new location, and thought that the path set in the setting is a better starting point than the last download.

But I could totally be convinced otherwise :)

Happy to change that and see if anyone else complains.

**Reply by drcjones** [reply]
Awesome reply!

I suggest:
- When selecting "Download" use the path from the download directory setting.
- When selecting "Download As" start at the last chosen folder (when exists). The scenarios when I use "Download As" instead of "Download" are usually when I'm downloading a series of videos in a row that I want in a specific folder. It would be frustrating to have to browse to that folder every time (and a bit tedious to temporarily change the default folder in the settings, then change it back to my normal one after).

Side note: The BETA Version 9.0.1.8 (beta) in Edge is not respecting the download directory setting when using "Download As". The regular Version 9.0.1.13 (stable) in Edge is working correctly.
Another note: When selecting "Convert Local" no versions seem to respect the download directory setting. I think it should.


Now to add on to @GrampaWildWilly 's post:
**In the list I'm seeing media I downloaded, but doesn't exist anymore.**
I was testing downloads with version 9 (in Edge {stable and beta} and FF beta) by downloading a bunch of copies of the same show using "Download As" and giving each a different name. Afterward I deleted all but 1 of the files. I renamed the 1 that I kept. The files I downloaded in a specific version of VDH are showing up in the popup (with options to play, deleted, or re-download), even though they were deleted from Windows Explorer. The renamed file isn't renamed in the popup so it's not playable. That being said, when I click the "clean all" button, or when I restart the browser, everything that was downloaded is removed from the list (as expected). So, I'm not sure if it makes sense or would even be worth the effort to keep the completed media in sync with the destination folder. However, I would like an option to not even show the completed media in the popup AND/OR to have it in a different list/location than the list of detected/downloading media (like a history tab or something like that).

**Comment by GrampaWildWilly** [comment]
Multi-line menu entries & width control of the drop down menu are intimately connected issues.  If you let us control the width of the menu, the possibility of fewer lines per entry improves.  Releases before VDH 9 had a setting for displaying entries on multiple lines.  I'm just asking for the restoration of an old behavior, really just an old setting.

For people who are not using smart naming, the issue of consistency between the VDH menu & the eventual target file name should be addressed.  Actually, you don't need to do anything except not use bold, all caps, & italics.  People who don't use smart naming will see the target file name on the VDH menu before launching a download.  That's how it's always worked.  That is in fact how it works now, but for the bold/all caps/italics issue.

For the most recently used directory, all versions of VDH before 9 caused the value of the Download Directory setting to change with each download.  Of course, if you did several consecutive downloads into the same directory, it didn't change.  But once a download went into a different directory, that became the new default Download Directory.  I'm just asking for the restoration of a long-standing prior VDH behavior.

Keeping the downloaded file entries on the VDH menu in synch with the contents of your disks is an issue Paul & I have already discussed.  It uses up a lot of overhead to do that automatically, dynamically.  For now, it's really on the user to just remove the entries for completed downloads yourself.  There is a setting "Automatically hide downloaded media."  Maybe you want to flip that one on.  You.  Maybe not other people.  But perhaps you.  There are a few functions available on the entries for completed downloads, among them download again, look at hit details, play (which just queues the file up into the VLC playlist).  Not to mention the reassurance such entries give by confirming the successful completion of a download.  So just deleting the entries is not & should not be a default or forced behavior of VDH.  The ways people use VDH vary a lot & there is probably no single behavior that would satisfy everybody.  I think VDH offers a rich enough set of behaviors that somewhere in that set, each user can find a personally satisfactory choice.

**Reply by GrampaWildWilly** [reply]
Correction.  The entries on the VDH menu for completed downloads do NOT offer a way to look at Hit Details.  In previous versions, that was true.  It is no longer.  I don't think this is much of a loss.  I hardly ever used that feature.  Hit Details are usually more useful on a variant before you download it, not after.
